### PR TITLE
New Climate entity logic

### PR DESCRIPTION
**Pull request by bob-tm**
- New climate logic, like original app
- Change entity names template. HA now can rename all device sensors with device rename.
- new device_class for duration and wifi sesnors
- Add retained flag for climate params (HA Restart unavailable bug)
- new sesnors:
  - binary_sensor_device_connected - device to grid connection status
  - sensor_active_setpoint - current temp to adjust from climate entity
  - sesnor_temperature_current - floor or room temp based on regulation type
  - sensor_temperature_target - target for climate entity

### CODE REVIEW

File: auto-discovery-templates/autogen-sensors.py [hunk 1]
```diff
@@ -26,6 +26,18 @@
     "on_time_total_kwh"
 ]
 
+
+topics_units = {
+    "sys_wifi_strength": "dBm",
+    "on_time_30_days": "s",
+    "on_time_7_days": "s",
+    "on_time_total": "s",
+    "on_time_30_days_kwh":"kwh",
```
Comment (stlk) [inline auto-discovery-templates/autogen-sensors.py]: HA expects the unit to be `kWh` otherwise they won't be visible in the Energy section.

**Review by stlk — COMMENTED** [review]
These changes are just what I've been looking for! I'm usually using thermostats on manual setting and that works very good. I've got only one note regarding `unit_of_measurement` HA expects.

File: auto-discovery-templates/autogen-sensors.py (hunk 1 above)
Comment (bob-tm) [reply auto-discovery-templates/autogen-sensors.py]: Tnx. Updated in my fork.

**Review by igor-podpalchenko — APPROVED** [review]
Merging to main

**Comment by igor-podpalchenko** [comment]
Check the new release. Pull the latest docker.

**Pull request merged**